dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/airflow/AirflowConnector.java

```diff
@@ -95,24 +117,123 @@ public void addTasksTo(@Nonnull List<? super Task<?>> out, @Nonnull ConnectorArg
     out.add(new DumpMetadataTask(arguments, FORMAT_NAME));
     out.add(new FormatTask(FORMAT_NAME));
 
+    ZonedIntervalIterable zonedIntervals = fromAirflowConnectorArguments(arguments);
+    Pair<ZonedDateTime, ZonedDateTime> dateRange = dateRange(arguments);
+
     // Airflow v1.5.0
-    addFullTable(out, "dag.csv", "select * from dag;");
-    addFullTable(out, "task_instance.csv", "select * from task_instance;");
+    addQueryTask(out, "dag.csv", "select * from dag;");
+    // todo think about multiple files
+    // addTableTaskWithIntervals(
+    //     out,
+    //     "task_instance",
+    //     "select * from task_instance ",
+    //     zonedIntervals,
+    //     (startDate, endDate) ->
+    //         String.format(
+    //             " where end_date >= CAST( '%s' as TIMESTAMP) and end_date < CAST( '%s' as
+    // TIMESTAMP) ;",
+    //             dateToSqlFormat(startDate), dateToSqlFormat(endDate)));
+    // // Airflow v1.6.0
+    // addTableTaskWithIntervals(
+    //     out,
+    //     "dag_run",
+    //     "select * from dag_run ",
+    //     zonedIntervals,
+    //     (startDate, endDate) ->
+    //         String.format(
+    //             " where end_date >= CAST( '%s' as TIMESTAMP) and end_date < CAST( '%s' as
+    // TIMESTAMP) ;",
+    //             dateToSqlFormat(startDate), dateToSqlFormat(endDate)));
+
+    addQueryTask(
+        out,
+        "task_instance.csv",
+        dateRange == null
+            ? "select * from task_instance;"
+            : "select * from task_instance "
+                + String.format(
+                    " where end_date >= CAST( '%s' as TIMESTAMP) and end_date < CAST( '%s' as TIMESTAMP) ;",
+                    dateToSqlFormat(dateRange.getLeft()), dateToSqlFormat(dateRange.getRight())));
+
     // Airflow v1.6.0
-    addFullTable(out, "dag_run.csv", "select * from dag_run;");
+    addQueryTask(
+        out,
+        "dag_run.csv",
```

Comment:
Looks like `dag_run.csv` and `task_instance.csv` use very similar SQL query. How about moving the logic that build query text to a separate function that takes (table_name, dateRange) as arguments to reduce duplicated code?

Reply:
I would say that all the sql queries are similar here, however I would not extract a new function for that reason. 

Downside of removing "duplication" is introduction coupling between the components. In the current case those tables are independent and queries may change in the future independently, so we do not need to couple the components.   

The question: `Will these queries change at the same time for the same reason in the future?`
The answer: `No, the tables are independent.`